### ISSUE TITLE
scope_where_ids_in for large collections

### ISSUE THREAD

**Issue by leemhenson** [toptal/chewy#138]
I've just been stepping through the code of what happens when I run ```chewy:update[my_index]``` and it looks a bit dicey to me, if you've got a large amount of documents to be indexed. For example, I have a  collection that is ~ 350k documents. It appears to be the case that chewy selects the ids of all documents into a 350k array. This works out to be nearly 30MB in size. It then sends this 30MB array over the wire as part of a single ```in``` query..... 

My database needs an :ambulance: .  :smile:

**Comment by pyromaniac** [comment]
Got it, I'm working on it :) I've just decided to use default scope in more obvious manner, so it would be possible to restrict the scope in indexed document for particular index completely. And actually I'be got an idea of subquery using for activerecord instead of fetching ids, but it seems like I would have the same effect in mongoid as well, so I've just decided to use subquery in AR and to fetch ids in mongoid in batches. Would implement this really soon, thanks for the report.
